### PR TITLE
Fix citation file

### PR DESCRIPTION
**Pull request by wbaccinelli**
The `cff-version` on the cff file was updated by `bump-my-version`. Also, the URL to the repo was not valid. 
The cff file has been fixed and a validation workflow has been added.

### CODE REVIEW

**Review by DaniBodor — APPROVED** [review]

**Comment by DaniBodor** [comment]
So we now in 2 separate places ensure that such mistakes don't happen again:
1) in the release workflow, we are checking that the CFF is valid.
2) the bump-version settings are more explicit to ensure only the correct lines get bumped

File: .bumpversion.toml
Comment (psomhorst) [inline .bumpversion.toml:1]: I'm not a fan of adding more files if configs can be gathered in pyproject.toml. Why extract these settings in particular, but not those of other tools?

File: CITATION.cff [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # YAML 1.2
 ---
-cff-version: "1.4.3"
+cff-version: 1.2.0
```
Comment (psomhorst) [inline CITATION.cff:3]: Good catch! Interesting that this is not a default skip by bump-my-version.

**Review by psomhorst — APPROVED** [review]
I can't comment on a file that is not changed, so:

actions/checkout@v3 in cffconvert.yml is deprecated and should be replaced with actions/checkout@v4. See warning here: https://github.com/EIT-ALIVE/eitprocessing/actions/runs/11627693839 and docs blogpost here: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/. 

Other than that: looks good! 

I assume this can just be merged to main without making a new release. 
Could you also merge it to develop? #314 may have a conflict with this PR, so I need to rebase that before merging with develop.

**Comment by wbaccinelli** [comment]
> I can't comment on a file that is not changed, so:
> 
> actions/checkout@v3 in cffconvert.yml is deprecated and should be replaced with actions/checkout@v4. See warning here: https://github.com/EIT-ALIVE/eitprocessing/actions/runs/11627693839 and docs blogpost here: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.

Sure, I'll update it

> I assume this can just be merged to main without making a new release. Could you also merge it to develop? #314 may have a conflict with this PR, so I need to rebase that before merging with develop.

I can also just merge to develop and the changes go with the next release. Also, a release will be necessary to update the reference in Zenodo, because it was broken and it is not up to date.

File: .bumpversion.toml
Comment (wbaccinelli) [reply .bumpversion.toml:1]: This is for @DaniBodor

**Comment by psomhorst** [comment]
> I can also just merge to develop and the changes go with the next release. Also, a release will be necessary to update the reference in Zenodo, because it was broken and it is not up to date.

In that case I think this PR could be used to make a patch release via https://github.com/EIT-ALIVE/eitprocessing/actions/workflows/release_github.yml, which will automatically be merged to develop.

File: .bumpversion.toml
Comment (DaniBodor) [reply .bumpversion.toml:1]: moved it back to pyproject.toml

**Comment by DaniBodor** [comment]
> > I can also just merge to develop and the changes go with the next release. Also, a release will be necessary to update the reference in Zenodo, because it was broken and it is not up to date.
> 
> In that case I think this PR could be used to make a patch release via https://github.com/EIT-ALIVE/eitprocessing/actions/workflows/release_github.yml, which will automatically be merged to develop.

@psomhorst , if everything is now ok, please go ahead and do the release.

**Pull request merged**